frontends/main/src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -57,14 +56,9 @@ class ErrorBoundaryHandler extends Component<
   }
 
   render() {
-    if (this.state.hasError) {
-      if (isForbiddenError(this.state.error)) {
-        return <ForbiddenPage />
-      } else {
-        return <NotFoundPage />
-      }
+    if (this.state.hasError && isForbiddenError(this.state.error)) {
+      return <ForbiddenPage />
```

Comment:
What is this change for? If I visit http://localhost:8062/c/unit/cow now I get 500 error.

Which... apparently I also get at https://next.rc.learn.mit.edu/c/unit/cow, though the UI shows the 404 page

Reply:
Opened https://github.com/mitodl/hq/issues/5734 for the current 500 situation

Reply:
The issue was that it was trapping all errors and displaying the 404 page, not the error.

We have a not-found.tsx in the app root to display the 404 page for routes not matched.

The request to a /c/unit/noexist (a matched route for a non existing channel) results in an error thrown while making the API request server side to fetch the channel details. We get a 404 from the API, though as it is not handled it produces a 500 at the browser with the error shown.

Here we need to handle and replay the 404. Errors thrown in server components or root layouts are not caught by the error boundary - this is likely fixed by adding an `app/error.ts` (server components) and `app/global-error.ts`, the widest catch-all ([info](https://nextjs.org/docs/app/building-your-application/routing/error-handling#handling-errors-in-root-layouts)).

Our current <ErrorBoundary> wraps children within the inner page - we should widen this for the whole page or remove is `app/error.ts` has the same effect.

Reply:
Let's handle #5734 in a dedicated error-handling PR?

Reply:
Yes - it's deserving of a separate issue. We need to:

- Reinstate 404 handling in the client component error boundary, but not for all errors (Check instance of AxiosError and status 404). This is to display a 404 if we open a not found resource drawer. I was finding this resulted in infinite retry (Is React Query retrying after errors?)

- Add a catch all error handler for server rendered content (`app/error.ts`) to render the 404 page or display the error.